src/Workspaces/Core/Portable/Execution/Serialization/HierarchicalChecksumObject.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Execution
+{
+    /// <summary>
+    /// HierarchicalChecksumObject indicates this type is collection of checksumObject types.
```

Comment:
it's a collection.. but doens't have any functionality that indicates it's a collection.  Why is this type needed?  Why can't subclasses of tihs just subclass ChecksumObject instead?